domain/crossmodelrelation/service/remoteapplication.go

```diff
@@ -126,6 +123,65 @@ func (s *Service) AddRemoteApplicationOfferer(ctx context.Context, applicationNa
 	return nil
 }
 
+// AddRemoteApplicationConsumer adds a new synthetic application representing
+// a remote relation on the consuming model, to this, the offering model.
+func (s *Service) AddRemoteApplicationConsumer(ctx context.Context, args AddRemoteApplicationConsumerArgs) error {
+	ctx, span := trace.Start(ctx, trace.NameFromFunc())
+	defer span.End()
+
+	// The synthetic application name is prefixed with "remote-" to avoid
+	// name clashes with local applications.
+	synthApplicationName := fmt.Sprintf("remote-%s", args.RemoteApplicationName)
+	if !application.IsValidApplicationName(synthApplicationName) {
+		return applicationerrors.ApplicationNameNotValid
+	}
+	if !uuid.IsValidUUIDString(args.OfferUUID) {
+		return internalerrors.Errorf("offer UUID %q is not a valid UUID", args.OfferUUID).Add(errors.NotValid)
+	}
+	if !uuid.IsValidUUIDString(args.RelationUUID) {
+		return internalerrors.Errorf("relation UUID %q is not a valid UUID", args.RelationUUID).Add(errors.NotValid)
+	}
+
+	// Construct a synthetic charm to represent the remote application charm,
+	// so we can track the endpoints it offers.
+	syntheticCharm, err := constructSyntheticCharm(synthApplicationName, args.Endpoints)
+	if err != nil {
+		return internalerrors.Capture(err)
+	}
+
+	remoteApplicationUUID, err := coreremoteapplication.NewUUID()
+	if err != nil {
+		return internalerrors.Errorf("creating remote application uuid: %w", err)
+	}
+
+	applicationUUID, err := coreapplication.NewID()
+	if err != nil {
+		return internalerrors.Errorf("creating application uuid: %w", err)
+	}
```

Comment:
This is the passed in `ApplicationToken`, which is a UUID. Use that, ensure it's correct, then pass it though to the state method.